tests/test_mau.py

```diff
@@ -232,19 +232,20 @@ def test_tracked_but_not_limited(self):
         self.reactor.advance(100)
         self.assertEqual(2, self.successResultOf(count))
 
-    def create_user(self, localpart, token=None):
-        request_data = json.dumps(
-            {
-                "username": localpart,
-                "password": "monkey",
-                "auth": {"type": LoginType.DUMMY},
-            }
-        )
+    def create_user(self, localpart, token=None, appservice=False):
+        request_data = {
+            "username": localpart,
+            "password": "monkey",
+            "auth": {"type": LoginType.DUMMY},
+        }
+
+        if appservice:
+            request_data["type"] = APP_SERVICE_REGISTRATION_TYPE
 
         channel = self.make_request(
             "POST",
             "/register",
-            request_data,
+            json.dumps(request_data),
```

Comment:
Not that you need to change it (as it matches the existing code), but heads up that `make_request` can actually just accept straight dictionaries these days.

Reply:
That's cool. Hopefully someone nice will come along and refactor the lot one day :)

Reply:
Oh except here I have changed it back  haven't i?

Reply:
Mmm, I don't think so. Before it was `request_data = json.dumps(...)`: https://github.com/matrix-org/synapse/pull/9548/files#diff-9f648c7d26b378d1bdae9e1d63e7a4c75d3645634680aa26e44dc959f99aa70dL236

But hooray for updating it anyways :tada:

Reply:
changed again